File: 프로그래머스/0/120846. 합성수 찾기/합성수 찾기.java
class Solution {
    private boolean check(int n){
        int cnt = 0;
        for(int i = 1; i*i <= n; i++){
            if(n % i == 0) {
                if(i * i == n) cnt++;
                else cnt += 2;
            }
        }
        return cnt >= 3;
    }
    public int solution(int n) {
        int ret = 0;
        for(int i = 1; i <= n; i++) if(check(i)) ret++;
        return ret;
    }
}

File: 프로그래머스/0/120846. 합성수 찾기/합성수 찾기.py
def count(n):
    cnt = 0
    for x in range(1, n+1):
        if x ** 2 > n: break
        if n % x == 0:
            if x ** 2 == n:
                cnt += 1
            else: cnt += 2
    return cnt

def solution(n):
    return sum(1 for x in range(4, n+1) if count(x) >= 3)